plugin/trino-duckdb/src/main/java/io/trino/plugin/duckdb/DuckDbClient.java

```diff
@@ -0,0 +1,305 @@
+    }
+
+    @Override
+    protected void renameTable(ConnectorSession session, Connection connection, String catalogName, String remoteSchemaName, String remoteTableName, String newRemoteSchemaName, String newRemoteTableName)
+            throws SQLException
+    {
+        execute(session, connection, format(
+                "ALTER TABLE %s RENAME TO %s",
+                quoted(catalogName, remoteSchemaName, remoteTableName),
+                quoted(catalogName, null, newRemoteTableName)));
+    }
+
+    @Override
+    public void commitCreateTable(ConnectorSession session, JdbcOutputTableHandle handle, Set<Long> pageSinkIds)
+    {
+        if (handle.getPageSinkIdColumnName().isPresent()) {
+            finishInsertTable(session, handle, pageSinkIds);
+        }
+        else {
+            renameTable(
+                    session,
+                    null,
+                    handle.getRemoteTableName().getSchemaName().orElse(null),
+                    handle.getTemporaryTableName().orElseThrow(() -> new IllegalStateException("Temporary table name missing")),
+                    handle.getRemoteTableName().getSchemaTableName());
+        }
+    }
+
+    @Override
+    public void addColumn(ConnectorSession session, JdbcTableHandle handle, ColumnMetadata column, ColumnPosition position)
+    {
+        if (!column.isNullable()) {
+            throw new TrinoException(NOT_SUPPORTED, "This connector does not support adding not null columns");
+        }
+        switch (position) {
+            case ColumnPosition.First _ -> throw new TrinoException(NOT_SUPPORTED, "This connector does not support adding columns with FIRST clause");
+            case ColumnPosition.After _ -> throw new TrinoException(NOT_SUPPORTED, "This connector does not support adding columns with AFTER clause");
+            case ColumnPosition.Last _ -> super.addColumn(session, handle, column, position);
+        }
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
+        if (mapping.isPresent()) {
+            return mapping;
+        }
+        switch (typeHandle.jdbcType()) {
+            case Types.BOOLEAN:
+                return Optional.of(booleanColumnMapping());
+            case Types.TINYINT:
+                return Optional.of(tinyintColumnMapping());
+            case Types.SMALLINT:
+                return Optional.of(smallintColumnMapping());
+            case Types.INTEGER:
+                return Optional.of(integerColumnMapping());
+            case Types.BIGINT:
+                return Optional.of(bigintColumnMapping());
+            case Types.FLOAT:
+                return Optional.of(realColumnMapping());
+            case Types.DOUBLE:
+                return Optional.of(doubleColumnMapping());
+            case Types.DECIMAL:
```

Comment:
Doesn't DuckDb supports precision via DatabaseMetadata.

Reply:
It supports precision but doesn't report scale. Let me leave a code comment.